Assets/VRTK/Scripts/VRTK_RealMovement.cs

```diff
@@ -0,0 +1,379 @@
+        public DirectionalMethod dirMethod = DirectionalMethod.Gaze;
+
+        [SerializeField]
+        [Tooltip("If we're using Smart Decoupling, all trackables must be within this degree threshold to change direction.")]
+        public float smartDecoupleThreshold = 30f;
+
+        // The cap before we stop adding the delta to the movement list.  This will help regulate speed.
+        [SerializeField]
+        [Tooltip("The max amount of movement we wish to register in the virutal world.  Decreasing this will increase acceleration, and vice versa.")]
+        public float sensitivity = 0.02f;
+
+        // The maximum number of updates we should hold to process movements.  The higher the number, the slower the acceleration/deceleration & vice versa.
+        private int averagePeriod = 60;
+        private float cameraLastYRotation = 0f;
+        
+        // Which tracked objects to use to determine amount of movement.
+        private List<Transform> trackedObjects = new List<Transform>();
+
+        private VRTK_PlayerPresence playerBody;
+        private Rigidbody rigidBody;
+
+        private GameObject controllerLeftHand;
+        private GameObject controllerRightHand;
+        private Transform headset;
+        private bool leftSubscribed;
+        private bool rightSubscribed;
+        private ControllerInteractionEventHandler touchpadPressed;
+        private ControllerInteractionEventHandler touchpadUp;
+
+        // List of all the update's movements over the average period.
+        private Dictionary<Transform, List<float>> movementList = new Dictionary<Transform, List<float>>();
+        private Dictionary<Transform, float> previousYPositions = new Dictionary<Transform, float>();
+
+        // Keeps track of the play area's last Y position.  This number is used to determine the delta position of the play area between updates.
+        // the delta y-position is then subtracted by all calculations to discount the movement of the play area from the trackable's movements. 
+        private float playAreaPreviousY = 0f;
+
+        // Used to determine the direction when using a decoupling method.
+        private Vector3 initalGaze = Vector3.zero;
+
+        // The current move speed of the player.  If Real Movement is not active, it will be set to 0.00f.
+        private float curSpeed = 0.00f;
+
+        // The current direction the player is moving.  If Real Movement is not active, it will be set to Vector.zero.
+        private Vector3 direction = new Vector3();
+
+        // True if Real Movement is currently engaged.
+        private bool active = false;
+
+        /// <summary>
+        /// Set your control options and modify the trackables to match.
+        /// </summary>
+        public void SetControlOptions(ControlOptions controlOptions)
+        {
+            this.controlOptions = controlOptions;
```

Comment:
don't include `this.` keyword, it makes VisualStudio moan because it's not required